Common/include/option_structure.hpp

```diff
@@ -174,9 +174,11 @@ enum AVERAGE_TYPE {
   AVERAGE_AREA = 1,     /*!< \brief Area-weighted average. */
   AVERAGE_MASSFLUX = 2  /*!< \brief Mass-flux weighted average. */
 };
+#ifdef ENABLE_MAPS
 static const map<string, AVERAGE_TYPE> Average_Map = CCreateMap<string, AVERAGE_TYPE>
 ("AREA", AVERAGE_AREA)
 ("MASSFLUX", AVERAGE_MASSFLUX);
+#endif
```

Comment:
The build times were increasing significantly as we divide the files, which does not make sense since the files are smaller. After some digging... the problem is the static maps declared in option_structure.hpp it seems that object code is generated for them in every file that includes option_structure.hpp (i.e. nearly every file in the project).
This adds a constant cost to any file (even if empty), as apparently the trickery that allows this convenient way of declaring the map is not cheap... also we have 129 maps...
We only need the declaration of the maps in config_structure.cpp and SU2_DOT.cpp, so my proposed solution is to guard the definition of each map, if you need them in a particular cpp define ENABLE_MAPS before including config_structure.hpp, if for some reason they are needed in an hpp ENABLE_MAPS should be undefined at the end to keep it from spreading.
One could of course move the maps to a separate file, but having them close to the corresponding enum is very convenient.
I will try to find a cleaner or at least less intrusive way of doing this.

Reply:
Update, the #ifdefs in every map can be avoided by making CCreateMap an empty class instead.
However that is not as good for compilation time as not doing any compilation...

Reply:
That seems indeed like a "problem" we should address. Thanks for pointing that out. However, I am not a big fan of the #ifdef solution.

Reply:
It is not a "problem" it is a **PROBLEM**, I know you guys over there have smart compilation tools that distribute compilation over the network, but for users with a normal computer or students with outdated laptops...

Reply:
The quotes were misleading. Don't get me wrong. I really think that this is a problem and we should do something about it.

Reply:
:+1: I do not like #ifdefs sprinkled all over the code either...
@talbring what about this solution:
```C++
#include <string>
#include <map>

//#define ENABLE_MAPS

using namespace std;

template<class T, class U>
struct EmptyMap {
    EmptyMap(initializer_list<pair<T,U> >) {}
};

#ifdef ENABLE_MAPS
    template<class T, class U>
    using MapType = map<T,U>;
    #define MakePair(a,b) {a,b},
#else
    template<class T, class U>
    using MapType = EmptyMap<T,U>;
    #define MakePair(a,b)
#endif


enum Options {
    OPTION_1,
    OPTION_2
};

static const MapType<string,int> myMap = {
    MakePair("Option_1", OPTION_1)
    MakePair("Option_2", OPTION_2)
};
```
It still involves the preprocessor and would require a small change of semantics, but I don't think there is a 0 cost solution with metaprogramming alone, the compiler still generates code for the initializer lists even is they are sunk into an empty map...

Reply:
I just tested in on a small file. It really makes a big difference whether you include option_structure.hpp or not .. How big is the overhead of your proposed solution? If it really helps, we can go for that one.

Reply:
Also, can you open a separate branch for this ?

Reply:
Compilation-wise it should be 0 as no code is generated, we'll know by testing.

> Also, can you open a separate branch for this ?

I was afraid you were going to ask xD. I can, it is the right way of doing it.